File: frontend/src/microservices/flight/objects/SearchSuggestionListPlannerResponse.ts
// 本文件定义 flight 模块的 `SearchSuggestionListPlannerResponse`，作为搜索建议列表响应并提供 JSON 编解码。

import type { SearchSuggestionPlannerResponse } from './SearchSuggestionPlannerResponse'

export type SearchSuggestionListPlannerResponse = {
  suggestions: SearchSuggestionPlannerResponse[]
}

export const searchSuggestionListPlannerResponseFromJson = (json: string): SearchSuggestionListPlannerResponse =>
  JSON.parse(json) as SearchSuggestionListPlannerResponse

export const searchSuggestionListPlannerResponseToJson = (value: SearchSuggestionListPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/SearchSuggestionListPlannerResponse.scala
// SearchSuggestionListPlannerResponse defines the flight suggestions planner response model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SearchSuggestionListPlannerResponse(
    suggestions: List[SearchSuggestionPlannerResponse]
)

object SearchSuggestionListPlannerResponse:
  given sourceEncoder: Encoder[SearchSuggestionListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[SearchSuggestionListPlannerResponse] = deriveDecoder

